hearthstone/player.py

```diff
@@ -285,8 +285,20 @@ def take_damage(self, damage: int):
     def redeem_gold_coin(self):
         if self.gold_coins >= 1:
             self.gold_coins -= 1
-            self.coins += 1
+            self.plus_coins(1)
 
     def gain_card(self, card: 'MonsterCard'):
         self.hand.append(card)
         self.check_golden(type(card))
+
+    def valid_board_index(self, index: 'BoardIndex') -> bool:
+        return 0 <= index < len(self.in_play)
+
+    def valid_hand_index(self, index: 'HandIndex') -> bool:
+        return 0 <= index < len(self.hand)
+
+    def valid_store_index(self, index: 'StoreIndex') -> bool:
+        return 0 <= index < len(self.store)
+
+    def plus_coins(self, amt: int):
+        self.coins = min(self.coins + amt, 10)
```

Comment:
We could make coins a private field, and only accessed through functions like this one.  Also see the way python properties work.

Reply:
So something like this?

`    def set_coins(self, amt: int):
        self._coins = min(amt, 10)`

`    def get_coins(self, amt: int):
        return self._coins`

Reply:
Whoops no amt argument in get_coins

Reply:
And whenever you buy a minion (for example), you'd need to do this?

`self.set_coins(self.get_coins() - self.minion_cost)`

This seems cumbersome, unless I'm missing something

Reply:
Yes that would be how it would work. However with python properties, you could just do

```
self.coins -= self.minion_cost
```
and it would work the same behind the scenes.

Reply:
So if I'm understanding properties correctly, we would have:

```
    @property
    def coins(self):
        return self.__coins
```

```
    @x.setter
    def coins(self, coins):
        self._coins = min(coins, 10)
```

and then 

`self.coins -= self.minion_cost`

would work.

Reply:
As an aside, can I push more changes onto this pull request, or do I need to close the pull request before editing?

Reply:
and whoops again @x.setter should be @coins.setter

Reply:
Yes exactly.  you should just push more commits to this branch, and they will show up in this PR.